Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Fleet/Driver.hs

```diff
@@ -237,15 +244,15 @@ postDriverFleetVerifyJoiningOtp merchantShortId opCity apiTokenInfo mbAuthId mbF
 postDriverFleetLinkRCWithDriver :: ShortId DM.Merchant -> City.City -> ApiTokenInfo -> Maybe Text -> Common.LinkRCWithDriverForFleetReq -> Flow APISuccess
 postDriverFleetLinkRCWithDriver merchantShortId opCity apiTokenInfo mbFleetOwnerId req = do
   checkedMerchantId <- merchantCityAccessCheck merchantShortId apiTokenInfo.merchant.shortId opCity apiTokenInfo.city
-  fleetOwnerId <- getFleetOwnerId apiTokenInfo.personId.getId mbFleetOwnerId
-  Client.callFleetAPI checkedMerchantId opCity (.driverDSL.postDriverFleetLinkRCWithDriver) fleetOwnerId req
+  requestorId <- getRequestorId apiTokenInfo mbFleetOwnerId
```

Comment:
here I think you need to use merchant based check for both requestorId, fleetOwnerId 
See getMbFleetOwnerAndRequestorIdMerchantBased,  getFleetOwnerAndRequestorIdMerchantBased functions from my PR